packages/api/src/controllers/stripe.ts

```diff
@@ -0,0 +1,720 @@
+          user.newStripeProductId !== "scale_1"
+        ) {
+          const items = await req.stripe.prices.list({
+            lookup_keys: products["prod_O9XuIjn7EqYRVW"].lookupKeys,
+          });
+          let subscription;
+
+          try {
+            subscription = await req.stripe.subscriptions.retrieve(
+              user.stripeCustomerSubscriptionId
+            );
+          } catch (e) {
+            console.log(`
+              Unable to migrate personal user - subscription not found for user=${user.id} email=${user.email} subscriptionId=${user.stripeCustomerSubscriptionId}
+            `);
+            continue;
+          }
+
+          if (subscription.status != "active") {
+            console.log(`
+              Unable to migrate personal user - user=${user.id} has a status=${subscription.status} subscription
+            `);
+            continue;
+          }
+
+          const subscriptionItems = await req.stripe.subscriptionItems.list({
+            subscription: user.stripeCustomerSubscriptionId,
+          });
+
+          subscription = await req.stripe.subscriptions.update(
+            user.stripeCustomerSubscriptionId,
+            {
+              billing_cycle_anchor: "now",
+              cancel_at_period_end: false,
+              items: [
+                ...subscriptionItems.data.map((item) => {
+                  // Check if the item is metered
+                  const isMetered =
+                    item.price.recurring.usage_type === "metered";
+                  return {
+                    id: item.id,
+                    deleted: true,
+                    clear_usage: isMetered ? true : undefined, // If metered, clear usage
+                    price: item.price.id,
+                  };
+                }),
+                ...items.data.map((item) => ({
+                  price: item.id,
+                })),
+              ],
+            }
+          );
+
+          await db.user.update(user.id, {
+            stripeCustomerId: user.stripeCustomerId,
+            stripeProductId: "prod_O9XuIjn7EqYRVW",
+            stripeCustomerSubscriptionId: subscription.id,
+          });
+
+          migratedUsers.push(user.email);
+
+          await sleep(200);
+        }
+      }
+    }
+
+    // Update cursor with newCursor after processing each batch of users
+    cursor = newCursor;
+  }
+
+  res.json(migratedUsers);
+});
+
+// Migrate Pro users to hacker with pay as you go
+app.post("/migrate-pro-users", async (req, res) => {
+  if (req.config.stripeSecretKey != req.body.stripeSecretKey) {
+    res.status(403);
+    return res.json({ errors: ["unauthorized"] });
+  }
+
+  const batchSize = 100;
+  let cursor = null;
+  let users = [];
+  let keepGoing = true;
+
+  let migratedUsers = [];
+  let firstUserId = null;
+
+  while (keepGoing) {
+    const [currentUsers, newCursor] = await db.user.find(
+      [sql`users.data->>'stripeProductId' = 'prod_1'`],
```

Comment:
Might make sense to check the `suspended` field of users as well (soccer)

Reply:
Tho keep in mind suspended users may be unsuspended. Maybe we want to migrate them anyway